pallets/stableswap/src/tests/withdraw.rs

```diff
@@ -0,0 +1,652 @@
+				withdraw_fee: Permill::from_percent(0),
+			},
+			InitialLiquidity {
+				account: ALICE,
+				assets: vec![
+					AssetLiquidity {
+						asset_id: asset_a,
+						amount: 100 * ONE,
+					},
+					AssetLiquidity {
+						asset_id: asset_b,
+						amount: 200 * ONE,
+					},
+					AssetLiquidity {
+						asset_id: asset_c,
+						amount: 100 * ONE,
+					},
+				],
+			},
+		)
+		.build()
+		.execute_with(|| {
+			let pool_id = get_pool_id_at(0);
+			let pool_account = AccountIdConstructor::from_assets(&vec![asset_a, asset_b, asset_c], None);
+
+			assert_ok!(Stableswap::add_liquidity(
+				Origin::signed(BOB),
+				pool_id,
+				vec![AssetLiquidity {
+					asset_id: asset_a,
+					amount: 100 * ONE,
+				},]
+			));
+
+			let shares = Tokens::free_balance(pool_id, &BOB);
+
+			assert_ok!(Stableswap::remove_liquidity_one_asset(
+				Origin::signed(BOB),
+				pool_id,
+				asset_b,
+				shares,
+			));
+
+			assert_balance!(BOB, asset_a, 100 * ONE);
+			assert_balance!(BOB, asset_b, 299_999_999_999_994u128);
+			assert_balance!(BOB, pool_id, 0u128);
+
+			assert_balance!(pool_account, asset_a, 200 * ONE);
+			assert_balance!(pool_account, asset_b, 100 * ONE + 6);
+		});
+}
+
+#[test]
+fn remove_liquidity_should_apply_correct_fee_when_removing_asset() {
+	let asset_a: AssetId = 1;
+	let asset_b: AssetId = 2;
+	let asset_c: AssetId = 3;
+
+	ExtBuilder::default()
+		.with_endowed_accounts(vec![
+			(BOB, 1, 200 * ONE),
+			(BOB, 2, 200 * ONE),
+			(BOB, 3, 200 * ONE),
+			(ALICE, 1, 200 * ONE),
+			(ALICE, 2, 300 * ONE),
+			(ALICE, 3, 200 * ONE),
+		])
+		.with_registered_asset("one".as_bytes().to_vec(), asset_a)
+		.with_registered_asset("two".as_bytes().to_vec(), asset_b)
+		.with_registered_asset("three".as_bytes().to_vec(), asset_c)
+		.with_pool(
+			ALICE,
+			PoolInfo::<AssetId> {
+				assets: vec![asset_a, asset_b, asset_c].try_into().unwrap(),
+				amplification: 100u16,
+				trade_fee: Permill::from_percent(0),
+				withdraw_fee: Permill::from_percent(10),
+			},
+			InitialLiquidity {
+				account: ALICE,
+				assets: vec![
+					AssetLiquidity {
+						asset_id: asset_a,
+						amount: 100 * ONE,
+					},
+					AssetLiquidity {
+						asset_id: asset_b,
+						amount: 200 * ONE,
+					},
+					AssetLiquidity {
+						asset_id: asset_c,
+						amount: 100 * ONE,
+					},
+				],
+			},
+		)
+		.build()
+		.execute_with(|| {
+			let pool_id = get_pool_id_at(0);
+			let pool_account = AccountIdConstructor::from_assets(&vec![asset_a, asset_b, asset_c], None);
+
+			assert_ok!(Stableswap::add_liquidity(
+				Origin::signed(BOB),
+				pool_id,
+				vec![AssetLiquidity {
+					asset_id: asset_a,
+					amount: 100 * ONE,
+				},]
+			));
+
+			let shares = Tokens::free_balance(pool_id, &BOB);
+
+			assert_ok!(Stableswap::remove_liquidity_one_asset(
+				Origin::signed(BOB),
+				pool_id,
+				asset_b,
+				shares,
+			));
+
+			assert_balance!(BOB, asset_a, 100 * ONE);
+			assert_balance!(BOB, asset_b, 288_003_020_713_436u128);
```

Comment:
I wonder if we could make the assertion of the fee more explicit.

Maybe getting it from the emitted event, putting it in a variable then substracting it from the balance? Something like

```
let fee = last_event().fee
assert_balance!(BOB, asset_b, 299_999_999_999_994u128 - fee);
```